### PR TITLE
Bug 1828457: bump(*): vendor update

### PR DESCRIPTION
**Pull request by mfojtik**
* openshift/api@27622da: Replace library-go references with build-machinery-go for CRD gen readme
* openshift/api@f5df921: Builds: Make a bunch of fields omitempty
* openshift/api@6c25f0f: config/v1/types_infrastructure.go: add service endpoints to aws platform spec and status
* openshift/api@1fc310b: operator/ingress: Add logging API
* openshift/api@2dadc74: config/v1/types_infrastructure.go: update the cloudConfig spec to define the generated one's location
* openshift/api@b7121e1: config/v1: add status subresource for infrstrastructure
* openshift/api@0f7ea19: bump to kube v1.18.2
* openshift/api@9e7957d: bump(build-machinery-go)
* openshift/build-machinery-go@8c085b9: Switch to go-bindata/go-bindata for consistency with upstream
* openshift/library-go@a39bf28: fileObserver: detect file existence when adding reactor
* openshift/library-go@b833546: Fix serving-certs race for controller's NewCommand*
* openshift/library-go@609e151: Fix ApplySecret for type changes
* openshift/library-go@e43e3d5: Fix error handling of status controller
* openshift/library-go@114736d: cmdbuilder: add WithVersion() that reports version metrics of binary
* openshift/library-go@eca6660: cmd: add WithVersion to StartController
* openshift/library-go@a20f1d9: Fix Apply{Deployment,Deployment} to copy input before mutatating
* openshift/library-go@42af545: Update ApplyDeployment to rely on a hash of the spec
* openshift/library-go@3b4f06f: Update ApplyDaemonSet to rely on a hash of the spec
* openshift/library-go@82e3f52: Bug 1826259: fix ingress CA cert location
* openshift/library-go@c1de486: Remove bitbucket.org/ww/goautoneg dependency
* openshift/library-go@4960ca0: Bump to kube v1.18.2
* openshift/library-go@d14c889: version: make sure to always have namespace when reporting version
* openshift/library-go@556325f: resourcesynccontroller: fix race in unit test
* openshift/library-go@b3ec7d2: vendor: Consolidate onto github.com/munnerz/goautoneg
* openshift/library-go@7813cff: staticpodstate: improve condition and event messages
* openshift/library-go@efe996a: bump(api,build-machinery-go)
* openshift/library-go@029d40b: Switch go go-bindata/go-bindata for consistency with upstream
* openshift/library-go@53b922b: make update
* openshift/library-go@e07b183: Update featuregate unit test to include APIPriorityAndFairness=true
* openshift/library-go@209722f: events: wire graceful broadcaster shutdown
* kubernetes/api@a9db9af: Update dependencies to v0.18.2 tag
* kubernetes/apimachinery@16eb159: Fix client watch reestablishment handling of client-side timeouts
* kubernetes/apiserver@de7df53: Update dependencies to v0.18.2 tag
* kubernetes/client-go@136e483: fix a bug where spn: prefix is unexpectedly added to kubeconfig apiserver-id setting
* kubernetes/client-go@dbcb4c3: Fix client watch reestablishment handling of client-side timeouts
* kubernetes/client-go@6b7c683: Update dependencies to v0.18.2 tag
* kubernetes/component-base@d8575c0: Update dependencies to v0.18.2 tag

### CODE REVIEW

File: vendor/github.com/openshift/library-go/pkg/operator/events/recorder_upstream.go [hunk 1]
```diff
@@ -27,20 +30,58 @@ type upstreamRecorder struct {
...
-	broadcaster := record.NewBroadcaster()
+	newRecorderForComponent := upstreamRecorder{
+		client:            r.client,
+		fallbackRecorder:  r.fallbackRecorder.WithComponentSuffix(componentName),
+		involvedObjectRef: r.involvedObjectRef,
+		shuttingDown:      r.shuttingDown,
+	}
+	// tweak the event correlator, so we don't loose important events.
+	broadcaster := record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{
+		BurstSize: 50,        // default: 25 (change allows a single source to send 50 events about object per minute)
```
Comment (deads2k) [inline vendor/github.com/openshift/library-go/pkg/operator/events/recorder_upstream.go:65]: You can't do this unconditionally.  This presents a high risk to controllers like the cluster-policy-controller. Make it a `With` option back in the controller builder.

**Comment by mfojtik** [comment]
@deads2k looking at events.json in artifacts, I think this is working as we want :-)

**Pull request merged**